File: app/src/test/java/com/permissionnanny/dagger/MockContextComponent.java
package com.permissionnanny.dagger;

import com.permissionnanny.ClientPermissionManifestReceiverTest;
import com.permissionnanny.ClientRequestReceiverTest;
import com.permissionnanny.ConfirmRequestActivityTest;
import com.permissionnanny.ConfirmRequestBinderTest;
import com.permissionnanny.UninstallReceiverTest;
import com.permissionnanny.missioncontrol.AppControlActivityTest;

/**
 * Dependency injection for tests.
 */
@ContextComponent.ContextScope
@dagger.Component(modules = {ContextModule.class}, dependencies = AppComponent.class)
public interface MockContextComponent extends ContextComponent {

    void inject(UninstallReceiverTest victim);

    void inject(ClientPermissionManifestReceiverTest victim);

    void inject(ClientRequestReceiverTest victim);

    void inject(ConfirmRequestBinderTest victim);

    void inject(AppControlActivityTest victim);

    void inject(ConfirmRequestActivityTest victim);
}

File: app/src/test/kotlin/com/permissionnanny/dagger/MockContextComponent.kt
package com.permissionnanny.dagger

import com.permissionnanny.ClientPermissionManifestReceiverTest
import com.permissionnanny.ClientRequestReceiverTest
import com.permissionnanny.ConfirmRequestBinderTest
import com.permissionnanny.UninstallReceiverTest

/**
 * Dependency injection for tests.
 */
@ContextComponent.ContextScope
@dagger.Component(modules = arrayOf(ContextModule::class), dependencies = arrayOf(AppComponent::class))
interface MockContextComponent : ContextComponent {

    fun inject(victim: UninstallReceiverTest)

    fun inject(victim: ClientPermissionManifestReceiverTest)

    fun inject(victim: ClientRequestReceiverTest)

    fun inject(victim: ConfirmRequestBinderTest)
}
